frame/instrumental-strategy-pablo/src/lib.rs

```diff
@@ -237,6 +247,29 @@ pub mod pallet {
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
+		/// Add vault_id to AssociatedVaults storage.
+		///
+		/// Emits [`AssociatedVault`](Event::AssociatedVault) event when successful.
+		#[pallet::weight(T::WeightInfo::associate_vault())]
+		pub fn associate_vault(
+			origin: OriginFor<T>,
+			vault_id: T::VaultId,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			let access_right = Self::admin_accounts(who).ok_or(Error::<T>::NotAdminAccount)?;
```

Comment:
I would create a function to abstract away the access control info. That way you can call `ensure!(caller_has_rights(who, AccessRights::AssocateVault), Error::<T>::UserDoesNotHaveCorrectAccessRight)` here.

Reply:
Resolved!